presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestJoinReorderorderingOptimizer.java

```diff
@@ -0,0 +1,109 @@
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.join;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.tableScan;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+
+public class TestJoinReorderorderingOptimizer
+{
+    private final LocalQueryRunner queryRunner;
+    private final Session reorderingEnabledSession;
+    private final Session reorderingDisabledSession;
+
+    public TestJoinReorderorderingOptimizer()
+    {
+        this.queryRunner = new LocalQueryRunner(testSessionBuilder()
+                .setCatalog("local")
+                .setSchema("tiny")
+                .build());
+
+        reorderingEnabledSession = queryRunner.getDefaultSession().withSystemProperty(SystemSessionProperties.JOIN_REORDERING, "true");
+        reorderingDisabledSession = queryRunner.getDefaultSession().withSystemProperty(SystemSessionProperties.JOIN_REORDERING, "false");
+
+        queryRunner.createCatalog(queryRunner.getDefaultSession().getCatalog().get(),
+                new TpchConnectorFactory(queryRunner.getNodeManager(), 1),
+                ImmutableMap.<String, String>of());
+    }
+
+    @Test
+    public void testNoReorderNeeded()
+    {
+        @Language("SQL") String sql = "select * from nation join region on nation.regionkey = region.regionkey";
+
+        PlanMatchPattern pattern =
+                anyTree(
+                        join(JoinNode.Type.INNER,
+                                ImmutableList.of(aliasPair("X", "Y")),
+                                anyTree(tableScan("nation")),
+                                anyTree(tableScan("region"))));
+
+        assertPlan(reorderingEnabledSession, sql, pattern);
+    }
+
+    @Test
+    public void testReorderNeeded()
+    {
+        @Language("SQL") String sql = "select * from region join nation on nation.regionkey = region.regionkey";
+
+        PlanMatchPattern pattern =
+                anyTree(
+                        join(JoinNode.Type.INNER,
+                                ImmutableList.of(aliasPair("X", "Y")),
+                                anyTree(tableScan("nation")),
+                                anyTree(tableScan("region"))));
+
+        assertPlan(reorderingEnabledSession, sql, pattern);
+    }
+
+    @Test
+    public void testReorderNeededButDisabled()
+    {
+        @Language("SQL") String sql = "select * from region join nation on nation.regionkey = region.regionkey";
+
+        PlanMatchPattern pattern =
+                anyTree(
+                        join(JoinNode.Type.INNER,
+                                ImmutableList.of(aliasPair("X", "Y")),
+                                anyTree(tableScan("region")),
+                                anyTree(tableScan("nation"))));
+
+        assertPlan(reorderingDisabledSession, sql, pattern);
```

Comment:
I would merge all these three tests into single tests as they use the same query and same pattern (for two cases)